### PR TITLE
Fix #1347 - Front-end for sending email to invite users to host team using URL.

### LINKED ISSUE #1347: Add feature to invite users to host team using URL
Very similar to #1346. 

## Deliverables

Add a feature to invite new team member to a participant team using a Unique URL. Also, don't forget to write unit tests. :) 

## Steps

1. **user1** enters the email of his teammate (let's say **user2**)
2. Send an email to **user2** (if he/she is registered on EvalAI) with a unique URL
3. **user2** clicks on the URL to accept the invitation to the team.
4. *user1* gets an email saying that "**user2** has accepted your invite to team <blah>" (we can iterate over the email message later) 

### Some important points to note

- [ ] Checks performed before inviting to the team should be similar to current invitation logic

### PR DESCRIPTION
**Pull request by isht3**
* Define new route for user's unique link.

     * Send $location.absUrl() to backend.

     * User's invite accepted feature.

Fixes #1347

### CODE REVIEW

**Comment by isht3** [comment]
@aka-jain Could you take a look at this PR? :smile:

**Comment by spyshiv** [comment]
@aka-jain Let me review this PR.
@isht3 will update you soon.

**Comment by isht3** [comment]
@spyshiv Thanks a tonne!

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 1]
```diff
@@ -325,22 +325,45 @@
                 parameters.url = 'hosts/challenge_host_teams/' + hostTeamId + '/invite';
                 parameters.method = 'POST';
                 parameters.data = {
-                    "email": result
+                    "email": result,
+                    "url": $location.absUrl(),
```
Comment (spyshiv) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please replace `$location.absUrl()` with `$state.href('web.challenge-host-teams', {}, {absolute: true})`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 2]
```diff
@@ -325,22 +325,45 @@
...
+            var team_hash = $stateParams.team_hash;
+            var email_hash = $stateParams.email_hash;
+            parameters = {};
+            parameters.url = 'hosts/invitation_host_team/' + team_hash + '/' + email_hash;
+            parameters.method = 'POST';
+            parameters.token = userKey;
+            parameters.data = {};
+            parameters.callback = {
+                onSuccess: function(response) {
+                    $location.path("web/dashboard");
```
Comment (spyshiv) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please replace `$location.path("web/dashboard")` with `$state.go("web.dashboard")`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 3]
```diff
@@ -325,22 +325,45 @@
...
+                    $rootScope.notify("success", response.data["message"]);
+                },
+                onError: function(response) {
+                    $location.path("web/dashboard");
```
Comment (spyshiv) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please replace `$location.path("web/dashboard")` with `$state.go("web.dashboard")`

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 4]
```diff
@@ -7,9 +7,9 @@
         .module('evalai')
         .controller('ChallengeHostTeamsCtrl', ChallengeHostTeamsCtrl);
 
-    ChallengeHostTeamsCtrl.$inject = ['utilities', 'loaderService', '$state', '$http', '$rootScope', '$mdDialog'];
+    ChallengeHostTeamsCtrl.$inject = ['$location', 'utilities', 'loaderService', '$state', '$stateParams', '$http', '$rootScope', '$mdDialog'];
```
Comment (spyshiv) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: Please remove `$location` dependency.

File: frontend/src/js/controllers/challengeHostTeamsCtrl.js [hunk 5]
```diff
@@ -325,22 +325,45 @@
...
+                    onError: function(response) {
+                        $rootScope.notify("error", "Couldn't invite. " + response.data.error + " Please try again.");
                     }
                 };
 
                 utilities.sendRequest(parameters);
             });
         };
 
+        if ($stateParams.email_hash){
```
Comment (spyshiv) [inline frontend/src/js/controllers/challengeHostTeamsCtrl.js]: __Bug__ : this `if(...)` is also executing in other routes that uses same controller (i.e. [http://127.0.0.1:8888/web/challenge-host-teams](http://127.0.0.1:8888/web/challenge-host-teams))

### How to solve this:
1. write a `vm.custom_function()` and move all the code that is inside `if(){...}`
```
vm.custom_function = function() {
        var team_hash = $stateParams.team_hash;
        .....
        .....
        .....
       // All the code that is inside if statement
     
}
```
2. Change route configuration and call `custom_function` from inline template
```
 var host_invitation = {
        name: "web.host-invitation",
        parent: "web",
        url: "/challenge-host-invitation/:team_hash/:email_hash",
        controller: 'ChallengeHostTeamsCtrl',
        controllerAs: 'challengeHostTeams',
        template: '<span ng-init="challengeHostTeams.custom_function()"><span>',
        title: 'Host Invitation',
        authenticate: true
 };
```
Done!!

**Comment by spyshiv** [comment]
@isht3: Done with the review. Please ping me in case of anything.

**Comment by isht3** [comment]
@spyshiv I have done the requested in this PR and also in #1415, thank you!

**Comment by spyshiv** [comment]
LGTM :)
Thanks @isht3 . This PR can be merged after #1489

**Comment by deshraj** [comment]
@isht3: lets get #1489 merged ASAP since this feature has been pending for a while.

**Comment by RishabhJain2018** [comment]
Closing this due to inactivity.

**Pull request closed without merging**